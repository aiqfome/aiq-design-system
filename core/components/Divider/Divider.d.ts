import React from 'react';
import { SpaceProps, ColorProps, LayoutProps } from 'styled-system';
export interface Props extends SpaceProps, ColorProps, LayoutProps {
    children?: any;
    color?: string;
}
export declare const Divider: React.FC<Props>;
//# sourceMappingURL=Divider.d.ts.map