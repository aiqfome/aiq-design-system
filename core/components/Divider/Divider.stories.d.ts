import React from 'react';
declare const _default: {
    component: React.FC<import("./Divider").Props>;
    title: string;
    decorators: any[];
};
export default _default;
export declare const Basic: React.FC;
export declare const withChildren: React.FC;
export declare const Height: React.FC;
//# sourceMappingURL=Divider.stories.d.ts.map