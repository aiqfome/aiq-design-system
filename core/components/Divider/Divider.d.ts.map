{"version":3,"file":"Divider.d.ts","sourceRoot":"","sources":["../../../src/components/Divider/Divider.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;AAEzB,OAAO,EAEL,UAAU,EAEV,UAAU,EAEV,WAAW,EACZ,MAAM,eAAe,CAAA;AAKtB,MAAM,WAAW,KAAM,SAAQ,UAAU,EAAE,UAAU,EAAE,WAAW;IAChE,QAAQ,CAAC,EAAE,GAAG,CAAA;IACd,KAAK,CAAC,EAAE,MAAM,CAAA;CACf;AASD,eAAO,MAAM,OAAO,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAsBnC,CAAA"}