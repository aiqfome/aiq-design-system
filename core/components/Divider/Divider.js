"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Divider = void 0;
const react_1 = __importDefault(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const styled_system_1 = require("styled-system");
const styled_components_1 = __importDefault(require("styled-components"));
const Flex_1 = require("../Flex");
const LineStyled = styled_components_1.default.hr `
  ${styled_system_1.layout}
  ${styled_system_1.space}
  ${styled_system_1.color}
  border: none;
`;
exports.Divider = (_a) => {
    var { children, height = '1px', color = 'mediumGrey' } = _a, props = __rest(_a, ["children", "height", "color"]);
    if (children) {
        return (react_1.default.createElement(Flex_1.Flex, { "data-testid": 'divider-wrapper', width: '100%', alignItems: 'center', justifyContent: 'center' },
            react_1.default.createElement(LineStyled, Object.assign({ height: height, backgroundColor: color }, props)),
            children,
            react_1.default.createElement(LineStyled, Object.assign({ height: height, backgroundColor: color }, props))));
    }
    return react_1.default.createElement(LineStyled, Object.assign({ height: height, backgroundColor: color }, props));
};
exports.Divider.propTypes = {
    children: prop_types_1.default.any,
    color: prop_types_1.default.string,
    height: prop_types_1.default.any
};
