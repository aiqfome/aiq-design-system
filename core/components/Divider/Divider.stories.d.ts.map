{"version":3,"file":"Divider.stories.d.ts","sourceRoot":"","sources":["../../../src/components/Divider/Divider.stories.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;;;;;;AAOzB,wBAIC;AAED,eAAO,MAAM,KAAK,EAAE,KAAK,CAAC,EAIzB,CAAA;AAED,eAAO,MAAM,YAAY,EAAE,KAAK,CAAC,EAMhC,CAAA;AAED,eAAO,MAAM,MAAM,EAAE,KAAK,CAAC,EAM1B,CAAA"}