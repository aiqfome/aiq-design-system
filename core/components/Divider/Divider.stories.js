"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Height = exports.withChildren = exports.Basic = void 0;
const react_1 = __importDefault(require("react"));
const addon_knobs_1 = require("@storybook/addon-knobs");
const Text_1 = require("../Text");
const Flex_1 = require("../Flex");
const Divider_1 = require("./Divider");
exports.default = {
    component: Divider_1.Divider,
    title: 'Divider',
    decorators: [addon_knobs_1.withKnobs]
};
exports.Basic = () => (react_1.default.createElement(Flex_1.Flex, { width: addon_knobs_1.text('width', '100%') },
    react_1.default.createElement(Divider_1.Divider, { width: addon_knobs_1.text('width', '100%') })));
exports.withChildren = () => (react_1.default.createElement(Divider_1.Divider, { width: addon_knobs_1.text('width', '100%'), color: addon_knobs_1.color('color', 'primary') },
    react_1.default.createElement(Text_1.Text, { paddingRight: 10, paddingLeft: 10 }, "Aiqfome!")));
exports.Height = () => (react_1.default.createElement(Divider_1.Divider, { margin: 10, height: addon_knobs_1.number('height', 5), color: addon_knobs_1.color('color', 'secondary') }));
