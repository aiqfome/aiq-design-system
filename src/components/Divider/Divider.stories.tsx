import React from 'react'
import { withKnobs, text, number, color } from '@storybook/addon-knobs'

import { Text } from '../Text'
import { Flex } from '../Flex'
import { Divider } from './Divider'

export default {
  component: Divider,
  title: 'Divider',
  decorators: [withKnobs as any]
}

export const Basic: React.FC = () => (
  <Flex width={text('width', '100%')}>
    <Divider width={text('width', '100%')} />
  </Flex>
)

export const withChildren: React.FC = () => (
  <Divider width={text('width', '100%')} color={color('color', 'primary')}>
    <Text paddingRight={10} paddingLeft={10}>
      Aiqfome!
    </Text>
  </Divider>
)

export const Height: React.FC = () => (
  <Divider
    margin={10}
    height={number('height', 5)}
    color={color('color', 'secondary')}
  />
)
