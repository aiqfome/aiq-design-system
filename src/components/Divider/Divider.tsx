import React from 'react'
import PropTypes from 'prop-types'
import {
  space,
  SpaceProps,
  color,
  ColorProps,
  layout,
  LayoutProps
} from 'styled-system'
import styled from 'styled-components'

import { Flex } from '../Flex'

export interface Props extends SpaceProps, ColorProps, LayoutProps {
  children?: any
  color?: string
}

const LineStyled = styled.hr<Props>`
  ${layout}
  ${space}
  ${color}
  border: none;

`

export const Divider: React.FC<Props> = ({
  children,
  height = '1px',
  color = 'mediumGrey',
  ...props
}) => {
  if (children) {
    return (
      <Flex width='100%' alignItems='center' justifyContent='center'>
        <LineStyled height={height} backgroundColor={color} {...props} />
        {children}
        <LineStyled height={height} backgroundColor={color} {...props} />
      </Flex>
    )
  }

  return <LineStyled height={height} backgroundColor={color} {...props} />
}

Divider.propTypes = {
  children: PropTypes.any,
  color: PropTypes.string,
  height: PropTypes.any
}
