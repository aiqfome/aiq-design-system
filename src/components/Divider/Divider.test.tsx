import React from 'react'
import renderer from 'react-test-renderer'

import { Divider } from './Divider'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Divider />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
