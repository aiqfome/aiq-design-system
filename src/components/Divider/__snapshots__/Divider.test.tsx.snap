// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<hr
  className="sc-gtssRu hFLlLZ"
  height="1px"
/>
`;
